code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1133,7 +1133,7 @@
 
 /mob/living/silicon/robot/MouseDrop_T(mob/living/M, mob/living/user)
 	. = ..()
-	if(!(M in buckled_mobs) && isliving(M))
+	if(!(M in buckled_mobs) && isliving(M) && ((user!=src)||!(src.get_active_held_item())))
```

Comment:
If you stick a return inside the if block right after `buckle_mob(M)`, and then move the `. = .. ()` to after the if block, you probably won't need the changes you have in mob.dm.